lib/python/qmk/tests/test_cli_commands.py

```diff
@@ -155,17 +155,18 @@ def test_json2c():
  * This file was generated by qmk json2c. You may or may not want to
  * edit it directly.
  */
+
 const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
     [0] = LAYOUT_ortho_1x1(KC_A)
 };
 
-#if defined(ENCODER_ENABLE) && defined(ENCODER_MAP_ENABLE)
-const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
-
-};
-#endif // defined(ENCODER_ENABLE) && defined(ENCODER_MAP_ENABLE)
 
 
+#ifdef OTHER_KEYMAP_C
+#    if __has_include_next(OTHER_KEYMAP_C)
```

Comment:
https://gcc.gnu.org/onlinedocs/cpp/Wrapper-Headers.html

Does this work reliably for `.c` files?

Reply:
As far as I know `gcc` makes no distinction based on file extension. Haven't had any issues with it so far.

Reply:
Plus, C++ has files without extensions 🤷

Reply:
```diff
diff --git a/quantum/quantum.c b/quantum/quantum.c
index 011f9d73e4..a815ecc3d4 100644
--- a/quantum/quantum.c
+++ b/quantum/quantum.c
@@ -16,6 +16,7 @@
 
 #include "quantum.h"
 
+#include_next "asdf.c"
 #ifdef BACKLIGHT_ENABLE
 #    include "process_backlight.h"
 #endif
```

```
quantum/quantum.c:19:2: error: #include_next in primary source file
```

Reply:
Odd, if I add `#error zzz` to the `keymap.c`:

```
Compiling: quantum/keymap_introspection.c                                                          In file included from ./.build/obj_chocv_tzarc_json/src/keymap.c:19,
                 from quantum/keymap_introspection.c:5:
/home/nickb/qmk_build/keyboards/chocv/keymaps/tzarc_json/keymap.c:20:2: error: #error zzz
   20 | #error zzz
      |  ^~~~~
 [ERRORS]
```

Reply:
Guessing it's because `keymap_introspection.c` is the "primary", which is including `keymap.c` from `.build` and that's counted as a non-primary and allowed to use `#include_next`.

Reply:
Now that I think about it, one scenario I haven't tried is having a keymap in the main qmk_firmware repo -- it's all been in userspace. It seems to use fully-resolved paths when located under userspace so I'll have to ensure it doesn't break things within main, assuming we want to migrate from `rules.mk` overall wherever possible.

Reply:
Turns out it's a fully-qualified path for userspace, and a qmk_firmware relative path for in-tree.
Have swapped it to just `#include OTHER_KEYMAP_C` without the existence check.